### DISCUSSION TITLE
`quarkus.grpc.server.use-separate-server` lead unexpected missing error grpc status

### DISCUSSION THREAD

**Discussion by spencercjh** [quarkusio/quarkus#30379] [Q&A] [answered]
When I disable `quarkus.grpc.server.use-separate-server` and use a common port for http and grpc server, I found that content of `io.grpc.Status` is missing in the response. I can only receive a `UNKNOWN` code response whether in my test codes or QuarkusDev.

![image](https://user-images.githubusercontent.com/29922079/212464324-f4ad0203-8ae9-41c8-8e18-bf8e6bd6b5b2.png)

![image](https://user-images.githubusercontent.com/29922079/212464331-ba4e2ff0-4e36-4f0d-a8d8-702e8f6164e8.png)

I can make sure that I correctly invoke `responseStreamObserver.onError` to return error to clients.

![image](https://user-images.githubusercontent.com/29922079/212464345-f4f3001d-1683-4b32-ba98-c08caaf3cc62.png)

cc @renyijiu

[a comment by an automated account is not shown]

**Comment by alesj** [comment]
Do you have a reproducer?
(or a test for this?)

**Comment by spencercjh** [comment]
@alesj I finally found problem. The exception passed to `responseObserver#onError` is build by `io.grpc.protobuf.StatusProto#toStatusRuntimeException` can't be processed correctly. When `quarkus.grpc.server.use-separate-server` is `false`, everything is OK.

reproducer:  https://github.com/spencercjh/reproducer-discussion-30379

**Comment by alesj** [comment, marked as answer]
Hmmm, this looks like an issue between our new Vert.x gRPC server and plain Netty based Channel.
If I set this
```
    clients:
      stub:
        use-quarkus-grpc-client: true
        host: localhost
        port: 8081
```
it works ...

@cescoffier @vietj ^^

**Reply by cescoffier** [reply]
Yes,

So, if you use `use-separate-server: true`, make sure your gRPC server does not use the SAME port of the main HTTP server (if enabled). If you don't the requests go to the main HTTP server and the server rejects the requests.  Note that I would expect a bind error in this case (need to check that). If the port are different, everything is fine.

If you use `use-separate-server:false`, then everything works as expected.

**Comment by alesj** [comment]
My note to my testing -- I was using latest local Quarkus build (999-SNAPSHOT).
Just to make sure new ExceptionHandler feature didn't already fixed this ...

I'll try to port tests over to Quarkus, and test all mix of combos -- new server+old client, old server + new client, ...

**Comment by alesj** [comment]
Here:
* https://github.com/alesj/quarkus/tree/i30379

3 tests fail ...

**Reply by cescoffier** [reply]
All fixed!

**Comment by spencercjh** [comment]
so, it's fixed now? In which release?

**Reply by cescoffier** [reply]
Yes, it's fixed. The safest it to take the latest one.
